### PR TITLE
[Galley] Adding ServiceEntry synthesis

### PR DESCRIPTION
**Pull request by nmittler**
Added a new custom projection that is subscribed to events for k8s Pods, Nodes, Services and Endpoints. These events are absorbed and do not become part of the snapshot. Instead, synthetic ServiceEntry resources are generated and become part of the snapshot.

Partially addresses #12276 and #10589

### CODE REVIEW

File: galley/pkg/runtime/projections/serviceentry/convert/convert.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+)
+
+// Annotations augments the annotations for the k8s Service.
+func Annotations(service resource.Entry, endpoints *resource.Entry) resource.Annotations {
+	out := make(resource.Annotations)
+	for k, v := range service.Metadata.Annotations {
+		out[k] = v
+	}
+
+	out[annotations.ServiceVersion] = string(service.ID.Version)
```
[the remark this answers is not in the document]
Comment (nmittler) [reply galley/pkg/runtime/projections/serviceentry/convert/convert.go]: @Nino-K 

The `ServiceEntry` version is used by MCP to indicate *something* changed.  With incremental MCP, I suspect you'll only receive resource if something actually changed.

Both service updates and endpoints will update the resource version of the `ServiceEntry`, which means you won't know which if you go by that alone.

[the remark this answers is not in the document]
Comment (Nino-K) [reply galley/pkg/runtime/projections/serviceentry/convert/convert.go]: Sure, if both endpoints and service change cause resource version bump then this logic makes total sense. I wasn't 100% sure if endpoint changes also bump the resource version.

**Comment by nmittler** [comment]
@ozevren PTAL. The galley integration test is now fixed ... I'm doing manual validation of the response using structpath.

File: pkg/test/framework/components/galley/snapshot.go [hunk 2]
```diff
@@ -43,9 +43,70 @@ func (*SnapshotObject) ProtoMessage()    {}
...
+		if len(actuals) != 1 {
+			return fmt.Errorf("expected 1 resource in snapshot, found %d", len(actuals))
+		}
+		return fn(actuals[0])
+	}
+}
+
+// NewSingleObjectSnapshotValidator creates a SnapshotValidatorFunc that ensures only a single object
+// is found in the snapshot.
+func NewSingleObjectSnapshotValidator(fn func(actual *SnapshotObject) error) SnapshotValidatorFunc {
```
Comment (ozevren) [inline pkg/test/framework/components/galley/snapshot.go]: Duplicate function?

**Comment by nmittler** [comment]
@costinm can you approve for the added dependencies?

**Comment by geeknoid** [comment]
Per TOC, we'd like to see this in 1.2

/approve
/lgtm

**Comment by nmittler** [comment]
@geeknoid @ozevren PTAL ... I had to resolve a merge issue.

**Pull request merged**